include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -204,7 +208,9 @@ class alignment_file_input_format<format_sam>
                       detail::is_type_specialisation_of_v<ref_offset_type, std::optional>,
                       "The ref_offset must be a specialisation of std::optional.");
 
-        auto stream_view = view::istreambuf(stream);
+        if (std::ranges::begin(stream_view) == std::ranges::end(stream_view)) // file has no records
```

Comment:
```suggestion
        if (std::ranges::empty(stream_view)) // file has no records
```
Should also work (http://eel.is/c++draft/range.prim.empty). 
But you can leave it as is if you want.

Reply:
I would like to keep this. begin and end are in the iterator header while empty is in the ranges header. as a follow up we will try to reduce the usage of the ranges header so this would need to be changed back then (if it is possible to be independent)